patches/minecraft/net/minecraft/item/BucketItem.java.patch

```diff
@@ -1,33 +1,104 @@
+          boolean flag = !material.func_76220_a();
+          boolean flag1 = material.func_76222_j();
+-         if (p_180616_2_.func_175623_d(p_180616_3_) || flag || flag1 || blockstate.func_177230_c() instanceof ILiquidContainer && ((ILiquidContainer)blockstate.func_177230_c()).func_204510_a(p_180616_2_, p_180616_3_, blockstate, this.field_77876_a)) {
+-            if (p_180616_2_.field_73011_w.func_177500_n() && this.field_77876_a.func_207185_a(FluidTags.field_206959_a)) {
++         if (p_180616_2_.func_175623_d(p_180616_3_) || flag || flag1 || blockstate.func_177230_c() instanceof ILiquidContainer && ((ILiquidContainer)blockstate.func_177230_c()).func_204510_a(p_180616_2_, p_180616_3_, blockstate, getFluid())) {
++            if (p_180616_2_.field_73011_w.func_177500_n() && getFluid().func_207185_a(FluidTags.field_206959_a)) {
+                int i = p_180616_3_.func_177958_n();
+                int j = p_180616_3_.func_177956_o();
+                int k = p_180616_3_.func_177952_p();
+@@ -126,8 +138,8 @@
+                for(int l = 0; l < 8; ++l) {
+                   p_180616_2_.func_195594_a(ParticleTypes.field_197594_E, (double)i + Math.random(), (double)j + Math.random(), (double)k + Math.random(), 0.0D, 0.0D, 0.0D);
+                }
+-            } else if (blockstate.func_177230_c() instanceof ILiquidContainer && this.field_77876_a == Fluids.field_204546_a) {
+-               if (((ILiquidContainer)blockstate.func_177230_c()).func_204509_a(p_180616_2_, p_180616_3_, blockstate, ((FlowingFluid)this.field_77876_a).func_207204_a(false))) {
++            } else if (blockstate.func_177230_c() instanceof ILiquidContainer && getFluid() == Fluids.field_204546_a) {
++               if (((ILiquidContainer)blockstate.func_177230_c()).func_204509_a(p_180616_2_, p_180616_3_, blockstate, ((FlowingFluid)getFluid()).func_207204_a(false))) {
+                   this.func_203791_b(p_180616_1_, p_180616_2_, p_180616_3_);
+                }
+             } else {
+@@ -136,7 +148,7 @@
+                }
+ 
+                this.func_203791_b(p_180616_1_, p_180616_2_, p_180616_3_);
+-               p_180616_2_.func_180501_a(p_180616_3_, this.field_77876_a.func_207188_f().func_206883_i(), 11);
++               p_180616_2_.func_180501_a(p_180616_3_, getFluid().func_207188_f().func_206883_i(), 11);
+             }
+ 
+             return true;
+@@ -147,7 +159,18 @@
     }
  
     protected void func_203791_b(@Nullable PlayerEntity p_203791_1_, IWorld p_203791_2_, BlockPos p_203791_3_) {
 -      SoundEvent soundevent = this.field_77876_a.func_207185_a(FluidTags.field_206960_b) ? SoundEvents.field_187627_L : SoundEvents.field_187624_K;
-+      SoundEvent soundevent = field_77876_a.getAttributes().getEmptySound();
-+      if(soundevent == null) soundevent = this.field_77876_a.func_207185_a(FluidTags.field_206960_b) ? SoundEvents.field_187627_L : SoundEvents.field_187624_K;
++      SoundEvent soundevent = getFluid().getAttributes().getEmptySound();
```

Comment:
Perhaps would be worth doing the same runtime redirect here as well to save a lot of patches